src/dbcp/data_mart/projects.py

```diff
@@ -15,20 +15,106 @@
+                    is_actionable,
+                    resource_clean,
+                    queue_status,
+                    queue_date AS date_entered_queue,
+                    interconnection_status_raw AS interconnection_status
+                FROM data_warehouse.gridstatus_projects
+                LEFT JOIN data_warehouse.gridstatus_resource_capacity USING (project_id)
+                LEFT JOIN data_warehouse.gridstatus_locations USING (project_id)
+            )
+            SELECT
+                sfip.state_name AS state,
+                cfip.county_name AS county,
+                gs.*,
+                1.0 AS frac_locations_in_county,
+                ncsl.permitting_type AS state_permitting_type FROM gs
+            LEFT JOIN data_warehouse.ncsl_state_permitting AS ncsl
+                on gs.state_id_fips = ncsl.state_id_fips
+            LEFT JOIN data_warehouse.state_fips AS sfip
+                ON gs.state_id_fips = sfip.state_id_fips
+            LEFT JOIN data_warehouse.county_fips AS cfip
+                ON gs.county_id_fips = cfip.county_id_fips
+
+        """
+        gs = pd.read_sql(query, con)
+
+    # TODO (bendnorman): How should we handle project_ids? This hack
+    # isn't ideal because the GS data warehouse and data mart project
+    # ids aren't consistent
+    max_lbnl_id = lbnl_non_isos.project_id.max() + 1
+    gs["project_id"] = list(range(max_lbnl_id, max_lbnl_id + len(gs)))
+
+    shared_ids = set(gs.project_id).intersection(set(lbnl_non_isos.project_id))
+    assert len(shared_ids) == 0, f"Found duplicate ids between GS and LBNL {shared_ids}"
+
+    fields_in_gs_not_in_lbnl = set(gs.columns) - set(lbnl.columns)
+    fields_in_lbnl_not_in_gs = set(lbnl.columns) - set(gs.columns)
+    assert (
+        not fields_in_gs_not_in_lbnl
+    ), f"These columns are in Grid Status but not LBNL: {fields_in_gs_not_in_lbnl}"
+    assert (
+        not fields_in_lbnl_not_in_gs
+    ), f"These columns are in LBNL but not Grid Status: {fields_in_lbnl_not_in_gs}"
+
+    return pd.concat([gs, lbnl_non_isos])
+
+
+def _get_and_join_iso_tables(
+    engine: sa.engine.Engine, use_gridstatus: bool = True
+) -> pd.DataFrame:
     """Get ISO projects.
 
     PK should be (project_id, county_id_fips, resource_clean), but county_id_fips has nulls.
 
     Note that this duplicates projects that have multiple prospective locations. Use the frac_locations_in_county
     column to allocate capacity and co2e estimates to counties when aggregating.
     Otherwise they will be double-counted.
+
+    Args:
+        engine: engine to connect to the local postgres data warehouse
+        use_gridstatus: use gridstatus data for ISO projects.
+
+    Returns:
+        A dataframe of ISO projects with location, capacity, estimated co2 emissions and state permitting info.
     """
     query = """
     WITH
     iso_proj_res as (
         SELECT
             proj.project_id,
+            proj.queue_id,
```

Comment:
I added `queue_id` because it is helpful when exploring and comparing the `iso_projects_long_format` table.